utils.bal

```diff
@@ -0,0 +1,641 @@
+    }
+    if (requestOptions?.ifMatchEtag != ()) {
+        request.setHeader(http:IF_MATCH, requestOptions?.ifMatchEtag.toString());
+    }
+    if (requestOptions?.enableCrossPartition == true) {
+        request.setHeader(IS_ENABLE_CROSS_PARTITION_HEADER, requestOptions?.enableCrossPartition.toString());
+    }
+    if (requestOptions?.isUpsertRequest == true) {
+        request.setHeader(IS_UPSERT_HEADER, requestOptions?.isUpsertRequest.toString());
+    }
+}
+
+//  Set the optional header specifying time to live.
+//  
+//  + request - http:Request to set the header
+//  + validationPeriod - the integer specifying the time to live value for a permission token
+//  + return - If successful, returns same http:Request with newly appended headers. Else returns error.
+//
+isolated function setExpiryHeader(http:Request request, int validationPeriod) returns error? {
+    if (validationPeriod >= MIN_TIME_TO_LIVE && validationPeriod <= MAX_TIME_TO_LIVE) {
+        request.setHeader(EXPIRY_HEADER, validationPeriod.toString());
+    } else {
+        return prepareUserError(VALIDITY_PERIOD_ERROR);
+    }
+}
+
+//  Map the json payload and necessary header values returend from a response to a tuple.
+//  
+//  + httpResponse - the http:Response or http:ClientError returned form the HTTP request
+//  + return - returns a tuple of type [json, ResponseMetadata] if sucessful else, returns error
+//
+isolated function mapResponseToTuple(http:Response httpResponse) returns @tainted [json, 
+        ResponseMetadata]|error {
+    json responseBody = check handleResponse(httpResponse);
+    ResponseMetadata responseHeaders = check mapResponseHeadersToHeadersRecord(httpResponse);
+    return [responseBody, responseHeaders];
+}
+
+//  Map the json payload and necessary header values returend from a response to a tuple.
+//  
+//  + httpResponse - the http:Response or http:ClientError returned form the HTTP request
+//  + return - returns a tuple of type [json, ResponseMetadata] if sucessful else, returns error
+//
+isolated function mapCreationResponseToTuple(http:Response httpResponse) returns @tainted [boolean, 
+        ResponseMetadata]|error {
+    boolean responseBody = check handleCreationResponse(httpResponse);
+    ResponseMetadata responseHeaders = check mapResponseHeadersToHeadersRecord(httpResponse);
+    return [responseBody, responseHeaders];
+}
+
+//  Handle sucess or error reponses to requests and extract the json payload.
+//  
+//  + httpResponse - http:Response or http:ClientError returned from an http:Request
+//  + return - If successful, returns json. Else returns error. 
+//
+isolated function handleResponse(http:Response httpResponse) returns @tainted json|error {
+    if (httpResponse.statusCode == http:STATUS_NO_CONTENT) {
+        //If status 204, then no response body. So returns empty json.
+        return {};
+    }
+    var jsonResponse = httpResponse.getJsonPayload();
+    if (jsonResponse is json) {
```

Comment:
you can remove this if else block when check is used since the program continue only if the response is json when check is used.